registrations/tasks.py

```diff
@@ -390,3 +393,164 @@ def run(
         connection.close()
 
 repopulate_metrics = RepopulateMetrics()
+
+
+class PullThirdPartyRegistrations(Task):
+
+    def get_or_create_identity(self, msisdn, details={}):
+
+        if not msisdn:
+            return None
+
+        msisdn = utils.normalize_msisdn(msisdn, '234')
+
+        params = {"details__addresses__msisdn": msisdn}
+        identities = utils.search_identities(params)
+
+        for identity in identities:
+            if details:
+                identity['details'].update(details)
+                utils.patch_identity(identity['id'], identity['details'])
+            return identity['id']
+
+        identity = {
+            'details': {
+                'addresses': {
+                    'msisdn': {
+                        msisdn: {'default': True}
+                    }
+                }
+            }
+        }
+        identity['details'].update(details)
+        identity = utils.create_identity(identity)['id']
+        return identity
+
+    def get_language(self, language):
+        return {'english': 'eng_NG',
+                'igbo': 'ibo_NG',
+                'pidgin': 'pcm_NG'}.get(language, language)
+
+    def get_msg_type(self, msg_type):
+        return 'audio' if msg_type == 'voice' else msg_type
+
+    def get_voice_times(self, time):
+        return {'9-11am': '9_11',
+                '2-5pm': '2_5',
+                '6-8pm': '6_8'}.get(time, time)
+
+    def get_voice_days(self, days):
+        return {'monday_and_wednesday': 'mon_wed',
+                'tuesday_and_thursday': 'tue_thu'}.get(days, days)
+
+    def get_receiver(self, receiver):
+        return {
+            "mother_and_father": "mother_father",
+            "mother_and_family": "mother_family",
+            "mother_and_friend": "mother_friend",
+        }.get(receiver, receiver)
+
+    def get_data(self):
+        url = settings.THIRDPARTY_REGISTRATIONS_URL
+        username = settings.THIRDPARTY_REGISTRATIONS_USER
+        password = settings.THIRDPARTY_REGISTRATIONS_PASSWORD
+
+        response = requests.get(url, verify=False, auth=(username, password))
```

Comment:
Is there a reason why we're not verifying the SSL certificate here? If they have issues, I think we should error on that. This is sensitive information, we should make sure that the connection is secure.

If they have issues with their certificates, we need to tell them to fix it.